src/inmanta/util.py

```diff
@@ -487,17 +493,32 @@ async def retry_limited(
     *args: object,
     **kwargs: object,
 ) -> None:
+    """
+    This function makes use of the INMANTA_RETRY_LIMITED_MULTIPLIER env variable. If set, INMANTA_RETRY_LIMITED_MULTIPLIER
+    serves as multiplier: The timeout given as argument becomes a 'soft limit' and the "soft limit' mutliplied by the
+    multiplier (form the env var) becomes a "hard limit". If the hard limit is reached before the wait condition is fulfilled
+    a Timeout exception is raised. If the wait condition is fulfilled before the hard limit is reached but after the soft
+    limit is reached, a different Timeout exception is raised. if the Env var is not set, then the soft and hard limit are
+    the same.
+    """
+
     async def fun_wrapper() -> bool:
         if inspect.iscoroutinefunction(fun):
             return await fun(*args, **kwargs)
         else:
             return fun(*args, **kwargs)
 
+    multiplier: int = int(os.environ.get("INMANTA_RETRY_LIMITED_MULTIPLIER", 1))
```

Comment:
I would also add a check to verify this value is `>=1` to prevent that someone uses this method incorrectly.